Makefile

```diff
@@ -47,13 +47,13 @@ CUDA_DIR  ?= $(or $(patsubst %/,%,$(dir $(patsubst %/,%,$(dir \
 AFLAGS = -fsanitize=address #-fsanitize=undefined -fno-omit-frame-pointer
 
 OPT    = -O -g
-CFLAGS = -std=c99 $(OPT) -Wall -Wextra -Wno-unused-parameter -fPIC -MMD -MP
-NVCCFLAGS = $(OPT)
+CFLAGS = -std=c99 $(OPT) -Wall -Wextra -Wno-unused-parameter -fPIC -MMD -MP -lstdc++
```

Comment:
`-lstdc++` is an `LDFLAGS` and presumably only exists if (a) C++ code is compiled in libCEED (depends on configuration) or (b) some other library needs it when using static libraries? Also, some platforms want to use `-lc++`. Probably best to just choose whether to use `LINK.c` or `LINK.cc` (these are implicitly defined by make; see `make -p -f/dev/null` for a complete list).

Reply:
As you know I'm really not comfortable with this kind of things, I had to use this flag to have successful compilation with the Cuda runtime API. I'm not attached to this flag, so if you have a solution that you think better, I have no problem using it.

Reply:
Actually, it seems that I needed this at some point but not any more...

Reply:
If I remove the `-lstdc++`, I see
```
$ lddtree build/t101-vec
build/t101-vec (interpreter => /lib64/ld-linux-x86-64.so.2)
    libceed.so => /home/jed/src/libceed/lib/libceed.so
        libocca.so => /home/jed/src/occa/lib/libocca.so
            libpthread.so.0 => /usr/lib/libpthread.so.0
            librt.so.1 => /usr/lib/librt.so.1
            libdl.so.2 => /usr/lib/libdl.so.2
            libOpenCL.so.1 => /usr/lib/libOpenCL.so.1
            libstdc++.so.6 => /usr/lib/libstdc++.so.6
            libgomp.so.1 => /usr/lib/libgomp.so.1
        libcudart.so.10.0 => /opt/cuda/lib64/libcudart.so.10.0
        libnvrtc.so.10.0 => /opt/cuda/lib64/libnvrtc.so.10.0
        libcuda.so.1 => /usr/lib/libcuda.so.1
            libnvidia-fatbinaryloader.so.415.25 => /usr/lib/libnvidia-fatbinaryloader.so.415.25
        libgcc_s.so.1 => /usr/lib/libgcc_s.so.1
    libm.so.6 => /usr/lib/libm.so.6
    libc.so.6 => /usr/lib/libc.so.6
```

which suggests that you are needing it only when OCCA uses static libraries. In that case, we could add `LDLIBS += -lstdc++` when OCCA is enabled. If you see this also with CUDA, then add if _either_ OCCA or CUDA is being used. Some compilers will produce noisy warnings if you use linker flags when only compiling. For example, your version produces
```
clang-7: warning: -Z-reserved-lib-stdc++: 'linker' input unused [-Wunused-command-line-argument]
```
for every compilation unit when I `make CC=clang`.